spec/features/proposals_spec.rb

```diff
@@ -203,6 +203,89 @@
         expect(page).not_to have_link("No comments", href: "#comments")
       end
     end
+
+    scenario "Shows clickable markers for related proposals by tag on proposal map", :js do
+      Setting["feature.map"] = true
+      create(:tag, :category, name: "culture")
+      create(:tag, :category, name: "sports")
+      proposal = create(:proposal, tag_list: "culture, sports", map_location: create(:map_location))
+      related_proposal = create(:proposal, tag_list: "culture", map_location: create(:map_location, latitude: 51.50))
+      create(:proposal, tag_list: "sports")
+
+      visit proposal_path(proposal)
+
+      within ".map_location" do
+        expect(page).to have_selector(".map-icon", count: 2)
+        expect(page).to have_selector(".map-icon.map-icon-related", count: 1)
+      end
+
+      find(".map-icon-related").click
+      within ".leaflet-popup-pane" do
+        click_link related_proposal.title
+      end
+
+      expect(page).to have_current_path proposal_path(related_proposal)
+    end
+
+    describe "Control layer to show/hide related proposals" do
+      scenario "Allow to hide related proposals", :js do
+        Setting["feature.map"] = true
+        create(:tag, :category, name: "culture")
+        create(:tag, :category, name: "sports")
```

Comment:
I think these two lines are not necessary; the `tag_list=` method automatically creates tags if they don't exist.